File: day1/part1.exs
{:ok, contents} = File.read("input.txt")

numbers =
  contents
  |> String.split("\n", trim: true)
  |> Enum.map(fn s ->
    {i, _} = Integer.parse(s)
    i
  end)

Enum.each(numbers, fn one ->
  case Enum.find(numbers, fn two -> one + two == 2020 end) do
    nil ->
      nil

    two ->
      IO.inspect(one * two)
  end
end)

File: day1/part1.rs
use std::fs;

fn main() {
    let contents = fs::read_to_string("input.txt").expect("Something went wrong reading the file");
    let numbers = contents.lines();
    let vec: Vec<&str> = numbers.collect();
    for one in &vec {
        for two in &vec {
            let o = match one.parse::<i32>() {
                Ok(o) => o,
                Err(_e) => -1,
            };
            let t = match two.parse::<i32>() {
                Ok(t) => t,
                Err(_e) => -1,
            };
            if o + t == 2020 {
                println!("{}", o * t);
            }
        }
    }
}
